packages/core/integration-tests/test/sourcemaps.js

```diff
@@ -1340,4 +1341,99 @@ describe('sourcemaps', function() {
       sourcePath,
     });
   });
+
+  it('carries sourcesContent from the original sources (tsx) through multiple transformations (babel and swc)', async () => {
+    let b = await bundle(
+      path.join(
+        __dirname,
+        'integration/sourcemap-original-sourcecontents-ts/index.tsx',
+      ),
+      {
+        defaultTargetOptions: {
+          shouldScopeHoist: true,
+        },
+      },
+    );
+
+    let filename = b.getBundles()[0].filePath;
+    let raw = await outputFS.readFile(filename, 'utf8');
+
+    let mapUrlData = await loadSourceMapUrl(outputFS, filename, raw);
+    if (!mapUrlData) {
+      throw new Error('Could not load map');
+    }
+    let map = mapUrlData.map;
+
+    let sourceMap = new SourceMap('/');
+    sourceMap.addVLQMap(map);
+    let sourceContent = map.sourcesContent[0];
+    let sourcePath = 'index.tsx';
+
+    checkSourceMapping({
+      map: sourceMap,
+      source: sourceContent,
+      generated: raw,
+      str: 'bar="bar"' /* from tsx: <App bar="bar" /> */,
+      generatedStr: 'bar: "bar"',
+      sourcePath,
+    });
+
+    checkSourceMapping({
+      map: sourceMap,
+      source: sourceContent,
+      generated: raw,
+      str: 'document.getElementById(',
+      sourcePath,
+    });
+  });
+
+  it('retains sourcesContent from the original sources from a large text file', async () => {
```

Comment:
This passed before and I'm not sure adds much value, but it verifies the fact that even though streams are not read to create sourceContent, this works since the sourcemaps code always buffers streams first, see: https://github.com/parcel-bundler/parcel/pull/6516#issuecomment-868896437